src/typing/typer.ml

```diff
@@ -2129,6 +2129,12 @@ and type_array_comprehension ctx e with_type p =
 	]) v.v_type p
 
 and type_return ?(implicit=false) ctx e with_type p =
+	let is_abstract_ctor = ctx.curfun = FunMemberAbstract && ctx.curfield.cf_name = "_new" in
+	let e =
+		match e with
+		| None when is_abstract_ctor -> Some (ECast((EConst(Ident "this"),p),None),p)
```

Comment:
I don't like doing this in syntax. We should instead handle this case below and generate the `texpr` node directly. IMO the whole problem with abstract constructors was brought about by implementing transformation like this in syntax, so let's make an effort to not make this even worse.

Reply:
> We should instead handle this case below and generate the texpr node directly.

Please, check the latest commit. Is that what you mean?

Reply:
Yes, but please use `get_this` instead of doing the lookup manually.